### DISCUSSION TITLE
Commit message suggestions by Copilot in VSCode should take the changes into account

### DISCUSSION THREAD

**Discussion by zoeyfrisart** [community/community#51577] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I noticed today that Copilot was creating suggestions for the commit message when writing a commit message inside of VSCode. While inherently, I think this is a fantastic idea. It would be more valuable if it considered the changes made in the commit. Since with the current approach the message it suggests does not at all take into account the changes made, for example for the code change:

```diff
if (count !== callCount.value) {
-  return;
+  return false;
}
```

The commit message it suggested was as follows:
<img width="326" alt="feat: MY-36233 change the way we handle the default value for the 'show' attribute of the 'show' directive. Now, if the 'show' attribute is not defined, the default value is 'true'. This is a breaking change, but is more consistent with the way we handle the 'show' attribute of the 'hide' directive" src="https://user-images.githubusercontent.com/19853448/229143137-65849340-3280-42a5-b1a8-3a02204a2315.png">

(for context, the value changed here does not affect a hide and show at all it affects the way we count the number of API calls made).

Only relevant discussion I was able to find was #48690, however, that discussion is relevant to add this, and this discussion is more feedback as to how this feature is implemented.

edit: For reference I am using copilot Labs, so this might be a feature that is still a work in progress*

**Comment by fahim9778** [comment]
How did you trigger this? Inside the commit box? I have Copilot Labs.

**Reply by zoeyfrisart** [reply]
Hi @fahim9778, I just randomly started getting this behaviour. It stopped doing this behaviour around a month after creating this discussion 🤷‍♀️

**Reply by MrPointer** [reply]
Probably a bit late to the party, but I suddenly started getting those copilot completions too, and then they've "magically" disappeared, which led me to this thread while searching for an answer - and I have found it! 🙂   
If anyone looks for the answer as well - There's a setting called `github.copilot.enable`, which is a map of "input sources" (I believe matches languages) and whether copilot is enabled for them or not. Specifically, there's the `scminput` item that controls whether copilot is enabled for commit messages.

**Reply by fuhrmanator** [reply]
> Specifically, there's the `scminput` item that controls whether copilot is enabled for commit messages.

Ironically (I know, it's not how LLMs work), Copilot didn't know about this when I asked it how to turn those off.

**Reply by rpresser** [reply]
When we did this recently, it added this at the bottom of the generated message:
`Co-authored-by: (my name), (my coworker's name)`

As it happened, we were pair programming over Teams at the moment this was done. So we started speculating that Copilot was spying on our Teams meeting. I thought it was more likely that Copilot just extracted all authors from the git log. Any evidence?

**Comment by bibblebabl** [comment]
@zoeyfrisart Now it's an experimental feature. You can trigger it in VSCode manually in Source Control `More Actions...` menu

<img width="354" alt="Screenshot 2023-10-25 at 17 17 17" src="https://github.com/community/community/assets/9008978/e2978273-c0d0-4ea2-a430-695b75938570">

**Reply by vincentbriglia** [reply]
Yours comes from GitLens though - not from Copilot - therefore requires an OpenAI API key, instead of CoPilot - afaik.

**Reply by bibblebabl** [reply]
Yes, you're right. Anyway, I have OpenAI so I will use it and wait for Copilot to bring this feature back.

**Comment by felipfr** [comment]
Is there any possibility that I can change the commit pattern to follow git-cz?

**Comment by vuon9** [comment]
How can I apply commitizen message? Is there a way to change the prompt for adapting commit message styling?

**Reply by yf-hk** [reply]
Please, we really need this 🙏

**Reply by Dhanush-git** [reply]
is this what you are looking for ?

<img width="828" alt="image" src="https://github.com/user-attachments/assets/76425367-f3ee-408a-b026-9a9538c9dab9" />

**Reply by vuon9** [reply]
It seems like what I need. I will try. Thanks so much @Dhanush-git !

**Comment by Bulletninja** [comment]
This seems to be still broken, even if I try @MrPointer 's `scminput` suggestion

**Reply by Bulletninja** [reply]
Other related discussions:
https://github.com/orgs/community/discussions/113557
https://github.com/orgs/community/discussions/102003
